openapi3/schema.go

```diff
@@ -930,13 +930,18 @@ func (schema *Schema) visitJSONString(value string, fast bool) (err error) {
 			schema.compiledPattern = cp
 		} else if v := schema.Format; len(v) > 0 {
 			// No pattern, but does have a format
-			re := SchemaStringFormats[v]
-			if re != nil {
-				cp = &compiledPattern{
-					Regexp:    re,
-					ErrReason: "JSON string doesn't match the format '" + v + " (regular expression `" + re.String() + "`)'",
+			f := SchemaStringFormats[v]
+			if f.regexp != nil {
```

Comment:
case switch on
* `f.regexp!=nil && f.callback==nil`
* `f.regexp==nil && f.callback!=nil`
* default: return an error "corrupted entry %q in SchemaStringFormats", v

Also: there needs to be a `f!=nil` check